framework/decode/dx12_object_info.h

```diff
@@ -303,7 +310,8 @@ struct D3D12ResourceInfo : DxObjectExtraInfo
     std::map<uint64_t, uint64_t>                       mapped_gpu_addresses;
     std::map<uint64_t, graphics::Dx12ShaderIdentifier> mapped_shader_ids;
 
-    D3D12_RESOURCE_DESC1 desc = {};
+    D3D12_RESOURCE_DESC1  desc = {};
+    D3D12_RESOURCE_STATES current_state{ D3D12_RESOURCE_STATE_COMMON };
```

Comment:
It needs to track the current state of resources for change state for copy resources.